### ISSUE TITLE
SDL2 OpenGL3 example not working on OSX 10.11.2 / Geforce 9400M

### ISSUE THREAD

**Issue by spate** [ocornut/imgui#468]
The SDL + OpenGL 3 example app isn't working on OSX 10.11.2 on my Geforce 9400M. The SDL + OpenGL (compatibility) example app is working fine. (Using top of tree.)

**Comment by Erkaman** [comment]
Same thing happens for me on OS X 10.10.3. sdl_opengl_example works fine, but in sdl_opengl3_example no windows are drawn:

<img width="1276" alt="screen shot 2016-01-03 at 22 40 34" src="https://cloud.githubusercontent.com/assets/602313/12080922/18eb6a1e-b26b-11e5-855b-514bfb5cab69.png">

**Comment by ocornut** [comment]
Issue was reported recently in #463 but not solved. I haven't personally tried this code on a Mac. 

Could you try: a/ the suggestions in that other thread? b/ disabling scissoring in the render function and see if it makes a difference?

Otherwise have a look at whatever error reporting SDL2 offers, it may give a clue. The core profile version looks definitively wrong to me but the poster of the other thread said it didn't fix the issue.

**Comment by Erkaman** [comment]
I have been able to solve the problem: Go to sdl_opengl3_example/main.cpp and replace this code:

```C++
	SDL_GL_SetAttribute(SDL_GL_DOUBLEBUFFER, 1);
	SDL_GL_SetAttribute(SDL_GL_DEPTH_SIZE, 24);
	SDL_GL_SetAttribute(SDL_GL_STENCIL_SIZE, 8);
	SDL_GL_SetAttribute(SDL_GL_CONTEXT_MAJOR_VERSION, 2);
	SDL_GL_SetAttribute(SDL_GL_CONTEXT_MINOR_VERSION, 2);
```

With this:

```C++
	SDL_GL_SetAttribute (SDL_GL_CONTEXT_FLAGS, SDL_GL_CONTEXT_FORWARD_COMPATIBLE_FLAG);
	SDL_GL_SetAttribute (SDL_GL_CONTEXT_PROFILE_MASK, SDL_GL_CONTEXT_PROFILE_CORE);
	SDL_GL_SetAttribute(SDL_GL_DOUBLEBUFFER, 1);
	SDL_GL_SetAttribute(SDL_GL_DEPTH_SIZE, 24);
	SDL_GL_SetAttribute(SDL_GL_STENCIL_SIZE, 8);
	SDL_GL_SetAttribute(SDL_GL_CONTEXT_MAJOR_VERSION, 3);
	SDL_GL_SetAttribute(SDL_GL_CONTEXT_MINOR_VERSION, 2);
```

The problem with the original code was that an OpenGL context of version 2.2(a compability profile) was created. However, the shaders in `imgui_impl_sdl_gl3.cpp` were using shader version 330, and this version can only be used in a core profile. Thus they would not compile.

So a core profile had to be created instead. However, you must be very careful when creating a core profile on mac. You have to set SDL_GL_CONTEXT_FLAGS to SDL_GL_CONTEXT_FORWARD_COMPATIBLE_FLAG _and_ set SDL_GL_CONTEXT_PROFILE_MASK to SDL_GL_CONTEXT_PROFILE_CORE. Otherwise, it will fail.

**Comment by ocornut** [comment]
Thanks for looking into it
It looks like `SDL_GL_SetAttribute (SDL_GL_CONTEXT_PROFILE_MASK, SDL_GL_CONTEXT_PROFILE_CORE);` was missing from my proposed fix in #463 
I will merge the fix in master shortly!

**Comment by ocornut** [comment]
Closing, thanks!
